tests/python/pants_test/engine/test_rules.py

```diff
@@ -248,20 +252,19 @@ def d_from_a(a: A) -> D:
     with self.assertRaises(Exception) as cm:
       create_scheduler(rules)
 
-    self.assert_equal_with_printing(
-      dedent(
-        f"""\
-        Rules with errors: 3
-          {fmt_rule(a_from_b_and_c)}:
-            Was not reachable, either because no rules could produce the params or because it was shadowed by another @rule.
-          {fmt_rule(a_from_c_and_b)}:
-            Was not reachable, either because no rules could produce the params or because it was shadowed by another @rule.
-          {fmt_rule(d_from_a)}:
-            Ambiguous rules to compute A with parameter types (B, C):
-              {fmt_rule(a_from_b_and_c)} for (B, C)
-              {fmt_rule(a_from_c_and_b)} for (B, C)
-       """
-      ).strip(),
+    self.assert_equal_with_printing(f"""\
+Rules with errors: 3
+  {fmt_rule(a_from_b_and_c)}:
+    Was not reachable, either because no rules could produce the params or because it was shadowed by another @rule.
+  {fmt_rule(a_from_c_and_b)}:
+    Was not reachable, either because no rules could produce the params or because it was shadowed by another @rule.
+  {fmt_rule(d_from_a)}:
+    Ambiguous rules to compute A with parameter types (B, C):
+      {fmt_rule(a_from_b_and_c)}
+for (B, C)
+      {fmt_rule(a_from_c_and_b)}
+for (B, C)
+""".strip(),
```

Comment:
Why remove the dedent? I personally find the formatting all the way to the left makes it a bit harder to scan the file because the individual test methods no longer look as visually distinct.

Reply:
I restored as many dedents as possible! I was unable to restore them all, because it didn't play nicely with nesting function calls within f-strings for some reason and I couldn't figure out how to make the spacing work. The exceptions are noted.